jetson/nav/README.md

```diff
@@ -1,156 +1,203 @@
+First we need to find the tag by driving in a spiral.
+The spiral is pre-generated and loaded from a file.
+The idea is that eventually we will spot the tag, then transition to the other sub state machine.
+Now we can handle the logic of discerning which of the left/right tag (in screen space) from the environment is the
+correct ID.
 
 ---
 
-<!----------------------------- Search ----------------------------->
-## Search
-Similar to the `gate_search/` folder, this folder for search logic contains a `searchStateMachine` object and files to define the waypoints for different types of searches. First we follow a square spiral outwards with points generated in spiralOutSearch.cpp, then if the search completes and the target is not found, we will move onto trying the lawnmower search and the spiral in search.
+## Gate Search
+
+[`gateStateMachine.cpp`](./gate_search/gateStateMachine.cpp)
 
+Sub state machine for gate search and traversal.
+Checking the source code is highly recommended as this has the most vector arithmetic.
+Basically we first line up with the gate, so we can cleanly drive straight through it.
+Sometimes this involves an extra step if both gate posts are in front of us (worst case).
+
+---
 
-#### `searchPointGeneration.py`
-This script is used to generate search patterns relative to the start point of the rover. The results are then written into `search/spiral_search_points.txt` as pairs of polar coordinates. The pairs are each "distance bearing". The reasoning for using polar coordinates is to use our pre-made utility function that generates points.
+## Obstacle Avoidance
 
-To use this script, you can simply run it and choose what you want it to do according to the prompts that you recieve. The first prompt is for the search type, and the rest are for the parameters. For testing, we currently use square spiral search with the parameters (13, 3).
+Not yet implemented.
 
 ---
 
-<!----------------------------- Vars and Utils ----------------------------->
-## Variables and Utilities
+## Utilities
 
-#### Nav State
-The nav state specifies what state of the state machine we are in. It is implemented as an enum (a C++ type) where named states are associated with a number behind the scenes.
+[`utilities.cpp`](./utilities.cpp) Misc. math, conversions between GPS and cartesian coordinates \
+[`filter.cpp`](./filter.hpp) Generic median + mean filter, useful for combining multiple readings into a more accurate
+one \
+[`cache.cpp`](./cache.hpp) Generic storage resilient to brief readings misses and false positives
 
-#### Auton State
-This is just a boolean hidden as a type called AutonState and it tells us if Auton is on or off. It is read from LCMs published by the GUI or simulator.
+---
 
-##### `utilities.cpp`
-Contains functions used commonly throughout auton code. 
+## State Diagram
 
+![image](https://user-images.githubusercontent.com/20666629/170621044-2055089d-cba9-4164-8472-e669dfc1ee66.png)
```

Comment:
can you add also auton being enabled in going from off to drive_waypoints/done

Reply:
can you add also in seeing posts/gates it's 2 posts with the correct ID, ik earlier gate was optimized to see just two posts but that was changed bc it was causing incorrect behavior

Reply:
@zwgold made the diagram so I would follow up with him about that

Reply:
@qhdwight Diagram should be updated now with the changes above.